src/include/symbol_table/symbole_table.h

```diff
@@ -0,0 +1,101 @@
+#ifndef JUCC_SYMBOL_TABLE_SYMBOL_TABLE_H
+#define JUCC_SYMBOL_TABLE_SYMBOL_TABLE_H
+
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+namespace jucc {
+
+struct Node {
+  /**
+   * Used to store the name of the identifier
+   * obtained during tokenization
+   */
+  std::string identifier_;
+  /**
+   * Used to store the data type of the identifier
+   * One of int or float
+   */
+  std::string data_type_;
+  /**
+   * Used to store the nesting level of scoping
+   * Deeper the nesting, higher the scope
+   *
+   *      ...
+   *      int a0_;
+   *      if ( condition ){
+   *          int a1_;
+   *          if ( second_condition ){
+   *              int a2_;
+   *              if ( third_condition){
+   *                      int a3_;
+   *                     ...
+   *              }
+   *          }
+   *      }
+   *      ...
+   *      a0_ has 0 level , a1_ has 1 level and so on ...
+   */
+  std::string nesting_level_;
+  /**
+   * Constructir for node class
+   */
+  Node(std::string it_, std::string dt_, std::string nt_) : identifier_(it_), data_type_(dt_), nesting_level_(nt_) {}
+};
+
+typedef struct Node Node;
```

Comment:
use using instead of typedef
modernize-use-using

Reply:
Ahh man